testsuite/gnat2goto/tests/derived_types/derived_types.adb

```diff
@@ -0,0 +1,11 @@
+with Private_Type; use Private_Type;
+procedure Derived_Types is
+   type My_Int is range -23 .. 23;
+   type My_Real is new Float;
+   type Derived_Priv is new Priv;
```

Comment:
So here we test that we can declare derived type of a private but cannot use it?

Reply:
We should be able to use it but without a call to a subprogram declared in the visible part of package private_type there is not much that we can do.
The assert statement should work but at the moment gnat2goto does not handle it.  I think the failure is something to do with type_conversions.